CamDetection.as

```diff
@@ -0,0 +1,146 @@
+void RenderMenuMain()
+{
+	if (!Setting_DebugShowActiveCamInMenu) {
+		return;
+	}
+	UI::BeginDisabled();
+	UI::MenuItem("Active Cam: " + tostring(g_activeCam));
+	UI::EndDisabled();
+}
+
+void UpdateActiveGameCam()
+{
+	g_activeCam = ActiveCam::None;
+
+	const Reflection::MwClassInfo@ tmTy = Reflection::GetType("CTrackMania");
+	const Reflection::MwMemberInfo@ gameSceneMember = tmTy.GetMember("GameScene");
+	uint GameCameraNodOffset = gameSceneMember.Offset + 0x10;
+
+	auto gameCameraNod = Dev::GetOffsetNod(GetApp(), GameCameraNodOffset);
+
+	// This is null in the menu and when cameras aren't being used.
+	if (gameCameraNod is null) {
+		return;
+	}
+
+	auto camModelNod = Dev::GetOffsetNod(gameCameraNod, 0x58);
+	auto camControlNod = Dev::GetOffsetNod(gameCameraNod, 0x68);
+
+	// Always 4 when backwards, and seemingly always 0 otherwise
+	bool isBackwards = Dev::GetOffsetUint32(gameCameraNod, 0xB0) == 0x4;
```

Comment:
Couple things: `gameCameraNod` is not a "nod", so that could be a bit confusing. In fact, this structure is `NGameCamera_SCamSys`, which is part of `NGameCamera_SMgr`.

This will crash the game on startup if any of these pointers are not valid (this function will be called every frame). That's not **necessarily** a bad thing (we have a killswitch at any time for these situations). But these offsets are deep enough that I'm not sure if I'm comfortable having them in there as it increases the maintenance burden quite a bit.

For starters, we might want to get the `SCamSys` structure using the visual managers (well defined in `VehicleState`, but might wanna move this to Openplanet's main API) instead of getting it from some random offset in the game.

Reply:
Ahh cool, yeah getting it via managers seems better (I just used the shortest pointer path). 

The .GameScene + 0x10 offset pointed to a `.GameCamera` object in MP4 (mb turbo too, cant remember), so I felt okay about it but if openplanet adds managers support then that definitely seems the best way.

I also found a pointer at `.GameTerminals[0] + 0x28` which might be an alternative (than .GameScene + 0x10). I haven't looked at NGameCamera_SMgr much, though I'm curious to see a buffer of NGameCamera_SCamSys under the mgr (I wonder if it's ever >1, might try splitscreen and check it out).

Reply:
I confirmed there are 2 SCamSys in splitscreen and the current method still worked fine.

Might it be possible to add manager support to the next openplanet version? If so, I can update to use that when there's an edge build.